upgrade/v110to120/upgrade.go

```diff
@@ -179,74 +209,93 @@ func translateStorageClassRecurringJobsToSelector(namespace string, kubeClient *
+		err = errors.Wrapf(err, "convert storageClass recurringJobs to recurringJobSelector failed")
+	}()
+	run.log = run.log.WithField("action", "convert-recurring-jobs-to-selectors")
+
+	if len(recurringJobIDs) == 0 {
+		run.log.Debug(upgradeLogPrefix + "Found 0 recurring job to convert to recurring job selector")
+		return
+	}
+
+	selectors := []recurringJobSelector{}
+	selectorIDs := []string{}
+	selectorParameter, ok := run.storageClass.Parameters["recurringJobSelector"]
 	if ok {
-		err = json.Unmarshal([]byte(scRecurringJobSelectorJSON), &scRecurringJobSelectors)
+		err = json.Unmarshal([]byte(selectorParameter), &selectors)
 		if err != nil {
-			return errors.Wrapf(err, "failed to unmarshal: %v", scRecurringJobs)
+			return errors.Wrapf(err, "failed to unmarshal: %v", selectorParameter)
 		}
-		for _, selector := range scRecurringJobSelectors {
-			scRecurringJobSelectorIDs = append(scRecurringJobSelectorIDs, selector.Name)
+		for _, selector := range selectors {
+			selectorIDs = append(selectorIDs, selector.Name)
 		}
 	}
 	for _, id := range recurringJobIDs {
-		if util.Contains(scRecurringJobSelectorIDs, id) {
+		if util.Contains(selectorIDs, id) {
 			continue
 		}
-		scRecurringJobSelectors = append(scRecurringJobSelectors, recurringJobSelector{
+		selectors = append(selectors, recurringJobSelector{
 			Name:    id,
 			IsGroup: false,
 		})
 	}
-	selectorJSON, err := json.Marshal(scRecurringJobSelectors)
+	selectorJSON, err := json.Marshal(selectors)
 	if err != nil {
-		return errors.Wrapf(err, "failed to marshal JSON %v", scRecurringJobSelectors)
+		return errors.Wrapf(err, "failed to marshal JSON %v", selectors)
 	}
-	log.Infof(upgradeLogPrefix+"Adding %v to recurringJobSelector", recurringJobIDs)
-	sc.Parameters["recurringJobSelector"] = string(selectorJSON)
-	log.Debug(upgradeLogPrefix + "Removing recurringJobs")
-	delete(sc.Parameters, "recurringJobs")
+	run.log.Infof(upgradeLogPrefix+"Adding %v to recurringJobSelector", recurringJobIDs)
+	run.storageClass.Parameters["recurringJobSelector"] = string(selectorJSON)
 
-	newStorageClassYAML, err := yaml.Marshal(sc)
-	storageClassCM.Data["storageclass.yaml"] = string(newStorageClassYAML)
-	logrus.Infof(upgradeLogPrefix+"Updating %v configmap", storageClassCM.Name)
-	if _, err := kubeClient.CoreV1().ConfigMaps(namespace).Update(context.TODO(), storageClassCM, metav1.UpdateOptions{}); err != nil {
-		return errors.Wrapf(err, "failed to update %v configmap", storageClassCM.Name)
+	run.log.Info(upgradeLogPrefix + "Removing recurringJobs")
+	delete(run.storageClass.Parameters, "recurringJobs")
+
+	logrus.Infof(upgradeLogPrefix+"Updating %v configmap", run.storageClassConfigMap.Name)
+	newStorageClassYAML, err := yaml.Marshal(run.storageClass)
+	run.storageClassConfigMap.Data["storageclass.yaml"] = string(newStorageClassYAML)
+	if _, err := run.kubeClient.CoreV1().ConfigMaps(run.namespace).Update(context.TODO(), run.storageClassConfigMap, metav1.UpdateOptions{}); err != nil {
```

Comment:
The StorageClass ConfigMap update (include unsetting `storageClass.Parameters["recurringJobs"]`) should  not be done before the RecurringJob CR creation.

Reply:
This method is called after recurringJob CR creation.
```
	if err := run.createRecurringJobCRs(); err != nil {
		return err
	}
	if err := run.convertToSelectors(recurringJobIDs); err != nil {
		return err
	}
```

Reply:
Oh sorry. The comparison shown by the GitHub review page confused me. The implementation here is right.